### DISCUSSION TITLE
Adding Id to sort and filter icon

### DISCUSSION THREAD

**Discussion by shivamBentley** [iTwin/iTwinUI#2236] [Q&A]
Hi, 

I am stuck in a situation where I want to add Id in filter and sort icon of table header. Is there any way to set id for those? 

![image](https://github.com/user-attachments/assets/7ad25946-09f0-4c34-b0c3-e10da81b1816)

**Comment by mayank99** [comment]
What's your use-case for such deep customizations? Table is built with lots of small elements all over the place; it would be difficult for us to provide the ability to customize every single thing.

**Reply by shivamBentley** [reply]
We are using pendo to track user activities. So, we want to ids for these.

**Reply by mayank99** [reply]
Does it only allow using `id`s? It would be better to find the element by its label (e.g. the Filter button is named "Filter").

![](https://github.com/user-attachments/assets/aca367c0-35d3-4145-8716-54a8f7b1fb84)
